chain/chain/src/types.rs

```diff
@@ -88,6 +93,8 @@ pub struct ApplyTransactionResult {
     pub total_gas_burnt: Gas,
     pub total_balance_burnt: Balance,
     pub proof: Option<PartialStorage>,
+    pub split_state_apply_results: Option<Vec<ApplySplitStateResult>>,
+    pub consolidated_state_changes: Option<ConsolidatedStateChanges>,
```

Comment:
Looking at the assert in `process_split_state_result`, is it true that at most one of these fields can be non-None? If thats the case, I suggest strongly-typing this: 

```rust
pub struct ApplyTransactionResult {
    pub trie_changes: WrappedTrieChanges,
    pub new_root: StateRoot,
    pub outcomes: Vec<ExecutionOutcomeWithId>,
    pub receipt_result: ReceiptResult,
    pub validator_proposals: Vec<ValidatorStake>,
    pub total_gas_burnt: Gas,
    pub total_balance_burnt: Balance,
    pub proof: Option<PartialStorage>,
    pub results: Option<Results>
}
enum Results {
  ApplySplitStateResult(Vec<ApplySplitStateResult>),
  ConsolidatedStateChanges(ConsolidatedStateChanges),
}
```

don't have enough context to suggest a good type name though!